### PR TITLE
Only show ideogram if related genes are found (SCP-2816)

### PR DESCRIPTION
**Pull request by eweitz**
This refines handling of Ensembl REST API outages by only showing the new related genes ideogram if results are found.  This resiliency enhancement preserves incremental rendering -- a significant performance optimization.

This satisfies SCP-2816.

### CODE REVIEW

**Review by bistline — APPROVED** [review]

File: app/assets/javascripts/scp-related-genes-ideogram.js [hunk 1]
```diff
@@ -103,6 +102,35 @@ function onPlotRelatedGenes() {
   window.SCP.log('ideogram:related-genes', props)
 }
 
+/**
+ * Set inner visibility of ideogram
+ *
+ * @param found Boolean Whether related genes have been found
+ */
+function setFoundRelatedGenes(found) {
```
Comment (devonbush) [inline app/assets/javascripts/scp-related-genes-ideogram.js:110]: Bootstrap has the built-in class 'hidden' that should simplify this and not require new css.  It looks like you want to start with the wrapper having the 'hidden' class, and then on finding related changes, toggle the 'hidden' class appropriately

File: app/assets/javascripts/scp-related-genes-ideogram.js [hunk 2]
```diff
@@ -103,6 +102,35 @@ function onPlotRelatedGenes() {
...
+  const innerFoundClass = (found) ? 'found-related-genes' : ''
+  const innerSelect = '#related-genes-ideogram-container #_ideogramInnerWrap'
+  document.querySelector(innerSelect).classList = innerFoundClass
+
+  if (found) {
+    const ideoContainer =
+      document.querySelector('#related-genes-ideogram-container')
```
Comment (devonbush) [inline app/assets/javascripts/scp-related-genes-ideogram.js:117]: I don't understand why you need to change classes on two different elements to make the ideogram visible.  Shouldn't showing/hiding the container be enough?

**Review by devonbush — CHANGES_REQUESTED** [review]
good quick work adding the event handler in ideogram (I presume that's why the version bumped).  Just some css questions

**Review by jlchang — APPROVED** [review]
Don't know enough javascript to be able to assess but I appreciate being included!

File: app/assets/javascripts/scp-related-genes-ideogram.js (hunk 1 above)
Comment (eweitz) [reply app/assets/javascripts/scp-related-genes-ideogram.js:110]: Good to know!  Done in 9e52773.

File: app/assets/javascripts/scp-related-genes-ideogram.js (hunk 2 above)
Comment (eweitz) [reply app/assets/javascripts/scp-related-genes-ideogram.js:117]: Good catch, done in 9e52773.

**Comment by eweitz** [comment]
This needs more work, based on Ensembl REST API behavior seen today.

When Ensembl is down, these changes work well.  When Ensembl is _slow but up_, though, these changes degrade UX compared to staging.  I set up a [monitoring dashboard](https://broadinstitute.slack.com/archives/CBEHTH601/p1602797026197300) to measure Ensembl availability and performance.

Anecdotally, Ensembl is more often slow than down, so what’s on staging seems overall better.  I'll try refining this to optimize around Ensembl availability scenarios, and close it if I don't have something reviewable by Monday.

What's on staging now is viable for production -- this should not block release.

**Comment by devonbush** [comment]
Ok, agree that we can hold off on releasing this.  Let's set aside some time to discuss the most robust solution

**Comment by eweitz** [comment]
I'll close this.  After a period of instability last week, monitoring indicates Ensembl has had excellent availability since Friday.

I plan to soon open a separate PR for logging to better answer "How often is Ensembl _up but slow_", among other analyses.

**Comment by eweitz** [comment]
Ensembl API is currently degraded.  Errors are sometimes thrown slowly (10+ seconds), but often quickly (~0.5 seconds).

**Pull request closed without merging**